docs: Style guide for APISIX documentation (#974)
iresty/iresty.github.io: website/docs/general/documentation-guide.md

---

> id: documentation-guide

Reviewer: ```suggestion
id: documentation-style-guide
```

title: Documentation style guide
keywords:
  - API gateway
  - APISIX
  - Apache APISIX
  - project documentations
description: Style guide for Apache APISIX documentation.
---

This document is a style guide for Apache APISIX documentation. This document is intended for contributors to APISIX's documentation and should be followed to ensure consistency.

To learn more about contributing, see [Contributing flow](contributor-guide.md).

> ## Tone, content and audience

Reviewer 2: ```suggestion
## Tone, Content and Audience
```

Author: So, for the style guide we are proposing, we will be using sentence case for all headings. It is much easier that way because we do not have to ensure title case everywhere and it is difficult to enforce title case everywhere (like in FAQ page).

Reviewer: After searching, other product documentation has different requirements on "Names of capitalization styles.". See https://en.wikipedia.org/wiki/Capitalization#Title_case.

For me, before deciding which style we should use, let's confirm the Document Style Guide Proposal first :)

Author: @juzhiyuan I guess the style guide was accepted and I have added the style guide in this PR.

Reviewer 2: I am fine with either way.

Reviewer: > I guess the style guide was accepted 

Hi @navendu-pottekkat, the style guide is accepted internally (you and me 😅), we need to invite more reviewers to reach our community's agreement. If no other comments, then it's ok.

For this https://github.com/apache/apisix-website/pull/974#discussion_r836980350, I mean our reviewers may start reviewing other changes than this one, so I left a message: https://github.com/apache/apisix-website/pull/974#issuecomment-1081311744.

As @yzeng25 also thinks it's ok to use both styles, then please resolve this comment. 😉

Author: Okay. Thanks.

- Aim for a voice and tone that's conversational, friendly, and respectful without being frivolous. Learn more on [Voice and tone](https://developers.google.com/style/tone).
- Write for accessibility. See [General do's and don'ts](https://developers.google.com/style/accessibility#general-dos-and-donts) to learn more.
- Identify the intended audience and write for them. Find the common ground for you and your target audience.

## Language

- Use second person while writing unless it isn't appropriate.
- Use active voice while writing.
- Use standard American spellings.
- Always place [conditional clauses before instructions](https://developers.google.com/style/clause-order).
- Use the following spellings for project specific entities:
  - Use Apache APISIX instead of APISIX when referencing the project in introductions and when referencing to the project community.
  - Use APISIX instead when referencing to the project inside documentation.
  - APISIX specific component names like Plugin and Route are always capitalized.
  - Use correct acronyms when necessary. Use the expanded word in first usage and proceed to use the acronym in the rest of the references. For example:

>     | ✅               | ❌               |

Reviewer: Center aligned?

Reviewer 3: I am not sure how to properly center align in docusauraus, but I have left suggestions for centering in GitHub markdown.

Author: You mean the style of the table should be center aligned? Use `---`?

Reviewer: Is it?
```
|: Column1 :|: Column 2 :|
```

Author: I looked at some projects and it seems like the left aligned tables are the default. Let's stick with that.

Is there any particular reason to suggest center alignment?

Reviewer: > I looked at some projects and it seems like the left aligned tables are the default. Let's stick with that.

OK to me.

>     | ---------------- | ---------------- |

Reviewer 3: ```suggestion
    | :--------------: | :--------------: |
```

For centering

Author: The normal `----` centers by default I guess.

Author: We can stick to the normal left aligned table as all the tables I find are left aligned. It would make it much easier to add new tables as well without having to point this out.

    | URL              | url              |
    | API              | api              |
    | APISIX Dashboard | Apisix dashboard |
    | gRPC             | GRPC/grpc        |

> ## Formatting, punctuation and organization

Reviewer 2: ```suggestion
## Formatting, Punctuation and Organization
```

Author: https://github.com/apache/apisix-website/pull/974#discussion_r836404167

  - Use [code blocks](https://docusaurus.io/docs/next/markdown-features/code-blocks) to display code. Add the file being modified as the title of the codeblock (for example, `coonf/config.yaml`). Use line highlighting to focus the attention of the use while explaining code.
  - Use [admonitions](https://docusaurus.io/docs/next/markdown-features/admonitions) to highlight important information. Use them as suggested below:
    - Note: for general information you want to stand out.
    - Tip: to give users bonus tips that build on top of the rest of the content. Can be ignored generally.
    - Important: to give extra emphasis to general information.
    - Warning: to highlight possibilities of loss of configuration or data.
    - Danger: user should be very careful about what is mentioned in the block. Use only in cases of maximum severity.
- Use tables to convey appropriate information easily. See [how they are used in FAQ](/docs/apisix/FAQ#what-is-the-difference-between-plugin-metadata-and-plugin-configs-in-apache-apisix).
- Use relative paths within markdown files `../xxx/xxx` instead of absolute paths.

> ### Directory structure

Reviewer 2: I guess this could be `##` instead of `###`, since it is not much related to "Formatting, punctuation and organization“ section.

Author: It could come under organisation. It shows how the documents are organised and how that organisation is configured in the configuration file. What do you think?

> The docs are organized as shown below. To create a new page, create a new file in the latest folder of the language you are writing the doc in.

Reviewer 2: ```suggestion
The documents are organized as shown below. To create a new page, create a new file in the latest folder of the language you are writing the document in.
```

```text
/docs
├── assets
│   ├── images
│   │   ├── xxxxx.png
│   └── other
│       └── xxxxx.xxx
├── en
│   └── latest
│       ├── doc1.md
│       ├── folder
│       │   └── doc2.md
│       │       └── folder2
│       │           └── doc3.md
│       └── config.json
├── es
│   └── latest
│       └── ...
└── zh
    └── latest

>         └── ...

Reviewer 2: This tree is somewhat inaccurate. We don't have the `es` part now. You can delete line 74-77 for now, see https://github.com/apache/apisix/tree/master/docs.

```

> ### Configuration file

Reviewer 2: I guess this could be `##` instead of `###`, since it is not much related to "Formatting, punctuation and organization“ section.

Author: https://github.com/apache/apisix-website/pull/974#discussion_r836402759

The configuration file is located in `/docs/<locale>/latest/config.json` where locale represents the locale code (language). Refer to [Locale Codes](https://www.science.co.il/language/Locale-codes.php) for more info. Note that locale is always all lowercase.

You can learn more about the sidebar from the [Docusaurus docs](https://v2.docusaurus.io/docs/next/sidebar).

> ```json

Reviewer: ```suggestion
```jsonc
```
